### PR TITLE
storageService: add timeout retry to async poll

### PR DESCRIPTION
**Pull request by sevein**
We've experienced random read timeouts when polling the Async SS
endpoint. This commit introduces a retry with up to four attempts and a
one second backoff.

We're planning to confirm this fix in QA before merging.

This is connected to https://github.com/archivematica/Issues/issues/280.

### CODE REVIEW

File: src/archivematicaCommon/lib/storageService.py [hunk 1]
```diff
@@ -28,6 +30,18 @@ class WaitForAsyncError(Error):
     pass
 
 
+# The default retry for idempotent endpoints.
+DEFAULT_RETRY = Retry(total=4, backoff_factor=1)
+
+# This retry can be used in POST requests where we want to make multiple
+# attempts in case of communication errors. It is advisable not to use this
+# retry unless you can afford having then endpoint run more than once.
```
Comment (jraddaoui) [inline src/archivematicaCommon/lib/storageService.py]: `then` -> `the`

**Review by jraddaoui — APPROVED** [review]
Nice!

**Comment by sevein** [comment]
This doesn't solve the problem, unfortunately. I'll leave it around because it can still be useful. See https://github.com/archivematica/Issues/issues/280 for more.

**Comment by sevein** [comment]
I'm going to close this for now. It's perfectly valid but it does not solve the problem reported by #280 so I don't want to add a new feature at this point.

**Pull request closed without merging**